lib/util/ssr.js

```diff
@@ -6,13 +6,45 @@
  */
 'use strict';
 
+let ssrEscapeFound = false;
+
 module.exports.isSSREscape = function isSSREscape(node) {
-    return (
-        (node.type === 'IfStatement' || node.type === 'ConditionalExpression') &&
-        (isMetaEnvCheck(node.test) || isWindowOrDocumentCheck(node.test))
-    );
+    if (node.type === 'IfStatement' || node.type === 'ConditionalExpression') {
+        ssrEscapeFound = false;
+        if (checkConditionalStatements(node.test) || isWindowOrDocumentCheck(node.test)) {
+            return true;
+        }
+    }
+
+    return false;
 };
 
+function checkConditionalStatements(test) {
+    let node = test;
+
+    // Recursive Case: If the node is a logical expression, check its left and right parts.
+    if (node.type === 'LogicalExpression' || node.type === 'BinaryExpression') {
+        checkConditionalStatements(node.left);
+        checkConditionalStatements(node.right);
```

Comment:
Rather than checking the `ssrEscapeFound` global, you could have `checkConditionalStatements` return a boolean, and then do a `|=` here.